src/Microsoft.DotNet.Cli.Utils/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe
+    /// </summary>
+    public static class EmbedAppNameInHost
+    {
+        private static string placeHolder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"; //hash value embedded in default apphost executable
+        private static byte[] bytesToSearch = Encoding.UTF8.GetBytes(placeHolder);
+        public static string EmbedAndReturnModifiedAppHostPath(
+            string appHostSourcePath,
+            string appHostDestinationDirectoryPath,
+            string appBinaryName)
+        {
+            var hostExtension = Path.GetExtension(appHostSourcePath);
+            var appbaseName = Path.GetFileNameWithoutExtension(appBinaryName);
+            var bytesToWrite = Encoding.UTF8.GetBytes(appBinaryName);
+            var destinationDirectory = Path.GetFullPath(appHostDestinationDirectoryPath);
+            var ModifiedAppHostPath = Path.Combine(destinationDirectory, $"{appbaseName}{hostExtension}");
```

Comment:
Should be camel-cased (copied from code that used a property, so this needs to be renamed).